File: Java/leetcode/sort-integers-by-the-number-of-1-bits/solution.java
import java.util.*;

class Solution {
    public int[] sortByBits(int[] arr) {
        Integer[] boxed = new Integer[arr.length];
        for (int i = 0; i < arr.length; i++) boxed[i] = arr[i];
        Arrays.sort(boxed, (a, b) -> {
            int pa = Integer.bitCount(a), pb = Integer.bitCount(b);
            if (pa != pb) return pa - pb;
            return a - b;
        });
        int[] res = new int[arr.length];
        for (int i = 0; i < arr.length; i++) res[i] = boxed[i];
        return res;
    }
}

File: C++/leetcode/sort-integers-by-the-number-of-1-bits/solution.cpp
class Solution {
public:
    vector<int> sortByBits(vector<int>& arr) {
        sort(arr.begin(), arr.end(), [](int a, int b) {
            int pa = __builtin_popcount(a), pb = __builtin_popcount(b);
            if (pa != pb) return pa < pb;
            return a < b;
        });
        return arr;
    }
};
